hoomd/md/half_step_hook.py

```diff
@@ -0,0 +1,30 @@
+# Copyright (c) 2009-2022 The Regents of the University of Michigan.
+# Part of HOOMD-blue, released under the BSD 3-Clause License.
+
+"""Perform user defined computations during the half-step of a \
+`hoomd.md.Integrator`.
+
+The HalfStepHook can be subclassed to define custom operations at the middle of
+each integration step. Examples of use cases include evaluating collective
+variables or biasing the simulation.
+"""
+
+from hoomd.md import _md
+
+
+class HalfStepHook(_md.HalfStepHook):
+    """HalfStepHook base class.
+
+    HalfStepHook provides an interface to perform computations during the
+    half-step of a hoomd.md.Integrator.
+    """
+
+    def update(self, timestep):
+        """Called during the half-step of a hoomd.md.Integrator.
```

Comment:
```suggestion
        """Called during the half-step of a `hoomd.md.Integrator`.
```
Link to the integtrator documentation.